tests/unit/config/test_phases.py

```
@@ -107,3 +109,148 @@ def test_get_phase_rate_stray_rate_attr_on_non_rate_phase_returns_none(
         )
         object.__setattr__(phase, "rate", 7.5)
         assert get_phase_rate(phase) is None
+
+
+class TestAdaptiveScalePhaseValidation:
+    def test_adaptive_scale_enabled_rejects_non_boolean_values(self) -> None:
+        with pytest.raises(ValidationError, match="adaptive_scale.enabled"):
```

Ruff (0.15.21)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)